lib/pacemaker/pcmk_output.c

```diff
@@ -2358,41 +2340,58 @@ result_code_xml(pcmk__output_t *out, va_list args)
 PCMK__OUTPUT_ARGS("cluster-verify", "pcmk_scheduler_t *", "int")
 static int
 cluster_verify_text(pcmk__output_t *out, va_list args) {
-
-    /* If there are verification errors, always print a statement about that, even if not requested */
-    
     pcmk_scheduler_t *scheduler = va_arg(args, pcmk_scheduler_t *);
     int section_opts = va_arg(args, int);
 
     pcmk__output_t *verify_out;
     int verify_rc;
+    int rc = pcmk_rc_ok;
+
+    (void)(verify_rc);
+    (void)(section_opts);
 
     pcmk__output_new(&verify_out, "none", NULL, NULL);
+
+    scheduler = pe_new_working_set();
+    scheduler->priv = verify_out;
+
     verify_rc = pcmk__verify(scheduler, verify_out, scheduler->input);
+
+    pe_free_working_set(scheduler);
     pcmk__output_free(verify_out);
 
     if (verify_rc == pcmk_rc_ok) {
         if (pcmk_is_set(section_opts, pcmk_section_verify)) {
+            PCMK__OUTPUT_LIST_HEADER(out, false, rc, "Cluster Summary");
```

Comment:
The way this macro works is that rc has to be initialized to pcmk_rc_no_output, and the macro will output the header the first time it's called and change rc to pcmk_rc_ok. Later calls will do nothing since the rc is changed. So, all calls for the same header have to use the same rc variable, which is why the header checks are all in the cluster-summary implementations.

This is a little different since we always want to output errors. Probably the easiest approach is to separate the verification itself from the output. We can always do the verification in the cluster-summary implementations, then output the message if the flag is set or there are errors.

Reply:
To make sure I'm understanding this correctly, do you think the verification (`pcmk__verify()`) call should happen in cluster-summary and the output (`PCMK__OUTPUT_LIST_HEADER()` and `out->list_item()`) should happen in cluster-verify?

Reply:
I was thinking cluster-summary should call `pcmk__verify()` and output the header, and cluster-verify would just output the message based on the rc. however at that point we might as well just do everything in cluster-summary and not have a separate message for cluster-verify